docs/src/service-workers-guide.md

```diff
@@ -0,0 +1,161 @@
+---
+id: service-workers-guide
+title: "Service Workers Guide"
+---
+
```

Comment:
Suggesting the following intro (rough sketch):

[Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) provide a browser-native way to handle requests made with the native [Fetch API (`fetch`)](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).

Depending on your testing needs, here are the scenarios supported by Playwright:
1. Test your page with a service worker. This works out of the box in all supported browsers.
2. Test your page without a service worker. Use `serviceWorkers: 'block'` option.
3. Route network traffic, whether it comes from a service worker or not. In Firefox and WebKit, use `serviceWorkers: 'block'` option to avoid service worker network traffic entirely. In Chromium, either block service workers or use [`BrowserContext.route()`].
4. Test your service worker implementation. Only in Chromium, use [`BrowserContext.serviceWorkers()`] to get access to the service worker and evaluate there.